### DISCUSSION TITLE
How can I enter easily a 2nd instance of xonsh without specifying the path

### DISCUSSION THREAD

**Discussion by halloleo** [xonsh/xonsh#5213] [Q&A] [answered]
I sometimes want to enter a 2nd instance of xonsh so that I can change environment variables and the like and return with a quick `exit` to the old standard xonsh. Two questions about this:

1. When i wnat enter a new instance,

        xonsh

    die not work. It just gives me "`<module 'xonsh' from '.../__init__.py'>`". I guess the problem is the command `xonsh` is interpreted in Python mode not in shell mode. So I help myself with

       /my/path/to/xonsh

    But the path is not on all systems the same. Is there a standard way to start a new instance?

2. When I am in the 2nd instance, how can I check that I am in a 2nd (or  3rd...) instance (without exiting)?

Thanks for any pointers!

**Comment by anki-code** [comment, marked as answer]
1. `<module 'xonsh' from '.../__init__.py'>`

I think you have `import xonsh` in `~/.xonshrc` or in some xontrib. If you run `del xonsh` or `xonsh --no-rc` the `xonsh` command will run xonsh. Dive into your RC or xontribs and remove `import xonsh` or replace it to `import xonsh as xonshlib`.

2. You can add something like this in your `~/.xonshrc`:

```xsh
$XONSH_NESTING_NUM = int(__xonsh__.env.get('XONSH_NESTING_NUM', 0))+1
```
Then:
```xsh
xonsh
$XONSH_NESTING_NUM
# 1
xonsh
$XONSH_NESTING_NUM
# 2
```

**Reply by halloleo** [reply]
Cool! Thanks @anki-code.

I _did_ have a `import xonsh` in my `.xonshrc`! I don't know anymore why I did so -- must have been from way early times when I explored some underlying features of Xonsh...

Neat trick with setting `$XONSH_NESTING` always from the last xonsh! 😄 So, `__xonsh__` refers to the last instance of xonsh?

**Reply by anki-code** [reply]
yes, `__xonsh__` is current `xonsh.built_ins.XonshSession`
